snap/gadget.go

```diff
@@ -69,6 +72,71 @@ type VolumeContent struct {
 	Unpack bool `yaml:"unpack"`
 }
 
+// GadgetConnect describes an interface connection requested by the gadget
+// between seeded snaps. The syntax looks like this:
+//
+// [<plug-snap-id1>|system]:plug [[<slot-snap-id>|system]:slot]
+//
+// "system" or omitting the snap-id indicates a system plug or slot.
+// Fully omitting the slot part indicates a system slot with the same name
+// as the plug.
+type GadgetConnect struct {
+	PlugSnapID string
+	Plug       string
+	SlotSnapID string
+	Slot       string
+}
+
+func parseSnapIDColonName(s string) (snapID, name string, err error) {
```

Comment:
I haven't looked into existing code but I feel like some day we may want a helper like this.

Reply:
Just to clarify, not requesting it right now.

Reply:
we have something similar in cmd/snap but not quite the same